### PR TITLE
Nginx es proxy version 2

### PR DESCRIPTION
**Pull request by joequery**
This PR does two things:

1) Creates an nginx role, including an `nginx_conf` module that allows us to easily enable a jinja2 file as an nginx configuration

2) Configure nginx to serve as a proxy in front of Elasticsearch. 

## Testing

Set `elasticsearch_nginx_proxy_enabled` to `true`.

Run the nginx role:

```shell
$ ansible-playbook -i staging.inventory -u vagrant --tags="site-nginx" --limit=vagrant-as-01 site-infrastructure.yml
```

Run the elasticsearch role:

```shell
$ ansible-playbook -i staging.inventory -u vagrant --tags="site-elasticsearch" --limit=vagrant-as-01 site-infrastructure.yml
```

SSH into the vm

```shell
$ vagrant ssh vagrant-as-01
```

Verify you can hit the local ES instance directly on port 9200

```
vagrant@vagrant-as-01:~$ curl "http://vagrant-as-01:9200"
{
  "name" : "vagrant-as-01-master_data",
  "cluster_name" : "elasticsearch",
  "version" : {
    "number" : "2.3.1",
    "build_hash" : "bd980929010aef404e7cb0843e61d0665269fc39",
    "build_timestamp" : "2016-04-04T12:25:05Z",
    "build_snapshot" : false,
    "lucene_version" : "5.5.0"
  },
  "tagline" : "You Know, for Search"
}
```

Verify when you try to hit the nginx proxy running on 9201 without credentials you get rejected

```
vagrant@vagrant-as-01:~$ curl "http://vagrant-as-01:9201"
<html>
<head><title>401 Authorization Required</title></head>
<body bgcolor="white">
<center><h1>401 Authorization Required</h1></center>
<hr><center>nginx/1.4.6 (Ubuntu)</center>
</body>
</html>
```

Verify the request through the proxy succeeds when the proper credentials are provided

```
vagrant@vagrant-as-01:~$ curl "http://user:password@vagrant-as-01:9201"
{
  "name" : "vagrant-as-01-master_data",
  "cluster_name" : "elasticsearch",
  "version" : {
    "number" : "2.3.1",
    "build_hash" : "bd980929010aef404e7cb0843e61d0665269fc39",
    "build_timestamp" : "2016-04-04T12:25:05Z",
    "build_snapshot" : false,
    "lucene_version" : "5.5.0"
  },
  "tagline" : "You Know, for Search"
}
```

Verify the nginx error and access logs are being written to

```
$ sudo tail /var/log/nginx/access.log
...
$ sudo tail /var/log/nginx/error.log
...
```

### CODE REVIEW

**Comment by andrewkcarter** [comment]
Instead of defining a dedicated group for the nginx role, why not define the role as a meta dependency of the elasticsearch role (and any other roles that want the proxy)?

**Comment by joequery** [comment]
That's a possibility. You might run into situations where nginx is installed but never used if the role doesn't opt-in to using it. Nginx is lightweight so that isn't too big of a deal.

**Comment by andrewkcarter** [comment]
I wonder if we can do a conditional in the meta yaml, that would solve it.

**Comment by andrewkcarter** [comment]
You can:
```
---
dependencies:
  - { role: 'nginx', when: elasticsearch_nginx_proxy_enabled is defined and elasticsearch_nginx_proxy_enabled }
```
I think I would prefer this solution over the groups simply because it eliminates the need to remember that you have to add the host to the nginx group, and it's easier to configure and manage via group and host vars without having to have special a inventory edit.

**Comment by joequery** [comment]
That's perfect. I'll get that change made. Thanks for the review.

File: testing.inventory [hunk 1]
```diff
@@ -16,6 +16,8 @@ repository_host=default
 #
 # Infrastructure and software topology
 #
+[nginx-nodes]
```
Comment (andrewkcarter) [inline testing.inventory]: Is this still required for anything?

**Review by andrewkcarter — APPROVED** [review]

**Pull request merged**